### PR TITLE
Move content from FAQ to proper places

### PR DESCRIPTION
**Pull request by axic**
Part of #1219.

Work in progress because there's still a lot to go. It can be reviewed and merged though.

### CODE REVIEW

**Comment by axic** [comment]
@chriseth can you review this? I'll probably add more content at some point, but maybe we should merge this now if it is good.

File: docs/contracts.rst [hunk 1]
```diff
@@ -556,13 +559,21 @@ In particular, the following operations will consume more gas than the stipend p
...
 .. warning::
     Contracts that receive Ether directly (without a function call, i.e. using ``send`` or ``transfer``)
     but do not define a fallback function
     throw an exception, sending back the Ether (this was different
     before Solidity v0.4.0). So if you want your contract to receive Ether,
     you have to implement a fallback function.
 
+.. warning::
+    A contract without a payable fallback function can receive Ether as a recipient of a `coinbase transaction` or
+    as a destination of a ``selfdestruct``. This is a design choice of the EVM and Solidity cannot work it around.
```
Comment (chriseth) [inline docs/contracts.rst]: `work around it`?

File: docs/contracts.rst [hunk 2]
```diff
@@ -556,13 +559,21 @@ In particular, the following operations will consume more gas than the stipend p
...
+
 .. warning::
     Contracts that receive Ether directly (without a function call, i.e. using ``send`` or ``transfer``)
     but do not define a fallback function
     throw an exception, sending back the Ether (this was different
     before Solidity v0.4.0). So if you want your contract to receive Ether,
     you have to implement a fallback function.
 
+.. warning::
+    A contract without a payable fallback function can receive Ether as a recipient of a `coinbase transaction` or
```
Comment (chriseth) [inline docs/contracts.rst]: double ```. Also perhaps add `(miner block reward)`

Comment (chriseth) [reply docs/contracts.rst]: Perhaps also add a sentence: A contract cannot react to such Ether transfers and thus also cannot reject them. This means that `this.balance` can be higher than the sum of some manually added "internal accounting".

File: docs/introduction-to-smart-contracts.rst [hunk 3]
```diff
@@ -57,6 +57,10 @@ and overwrite your number, but the number will still be stored in the history
 of the blockchain. Later, we will see how you can impose access restrictions
 so that only you can alter the number.
 
+.. note::
+    All identifiers (contract names, function names and variable names) are restricted to
+    the ASCII character set. It is possible to store UTF-8 encoded data in string variables.
```
Comment (chriseth) [inline docs/introduction-to-smart-contracts.rst:62]: We should add a note somewhere stating that `UTF-8` encoded strings might look identical in an editor but can still be different at the byte level.

File: docs/types.rst [hunk 4]
```diff
@@ -127,6 +127,24 @@ the function ``call`` is provided which takes an arbitrary number of arguments o
 
 ``call`` returns a boolean indicating whether the invoked function terminated (``true``) or caused an EVM exception (``false``). It is not possible to access the actual data returned (for this we would need to know the encoding and size in advance).
 
+It is possible to adjust the supplied gas with the ``.gas()`` modifier::
+
+    namReg.call.gas(1000000)("register", "MyName");
+
+Similarly, the supplied Ether value can be controlled too::
+
+    nameReg.call.value(1 eth)("register", "MyName");
```
Comment (chriseth) [inline docs/types.rst]: I think it should be `1 ether`

File: docs/types.rst [hunk 5]
```diff
@@ -127,6 +127,24 @@ the function ``call`` is provided which takes an arbitrary number of arguments o
...
+
+Lastly, these modifiers can be combined. Their order does not matter::
+
+    nameReg.call.gas(1000000).value(1 eth)("register", "MyName");
```
Comment (chriseth) [inline docs/types.rst]: also here.

File: docs/types.rst [hunk 6]
```diff
@@ -169,6 +187,10 @@ Members:
 
 * ``.length`` yields the fixed length of the byte array (read-only).
 
+.. note::
+    It is possible to use an array of bytes as ``byte[]``, but it is wasting a lot of space, 31 bytes every every element,
```
Comment (chriseth) [inline docs/types.rst]: duplicate `every`

File: docs/units-and-global-variables.rst [hunk 7]
```diff
@@ -73,6 +73,18 @@ Block and Transaction Properties
     This includes calls to library functions.
 
 .. note::
+    Do not rely on ``block.timestamp`` and ``block.blockhash`` as a source of randomness,
```
Comment (chriseth) [inline docs/units-and-global-variables.rst]: Please also mention `now`

File: docs/units-and-global-variables.rst [hunk 8]
```diff
@@ -73,6 +73,18 @@ Block and Transaction Properties
...
+    unless you know what you are doing.
+
+    The timestamp is supplied by miners and the content (i.e. transactions included) in the
```
Comment (chriseth) [inline docs/units-and-global-variables.rst]: Perhaps simplify this to
```
Both the timestamp and the block hash can be influenced by miners to some degree. Miners can for example run a casino payout function on a chosen hash and just re-try a different hash if they did not receive any money.

The current block ...
```
```

File: docs/contracts.rst (hunk 2 above)
Comment (axic) [reply docs/contracts.rst]: It is single "`" on purpose, because that is italic.

File: docs/units-and-global-variables.rst (hunk 7 above)
Comment (axic) [reply docs/units-and-global-variables.rst]: It is in the list above this note, but can add it.

**Comment by axic** [comment]
@chriseth updated

File: docs/introduction-to-smart-contracts.rst (hunk 3 above)
Comment (axic) [reply docs/introduction-to-smart-contracts.rst:62]: Added as a warning block below.

**Review by chriseth — APPROVED** [review]

**Pull request merged**